bob-backend/src/pearl/group.rs

```diff
@@ -76,19 +76,20 @@ impl Group {
         debug!("{}: save holders to group", self);
         self.add_range(holders).await;
         debug!("{}: start holders", self);
-        self.run_pearls().await
+        self.run_pearls(pp).await
     }
 
     pub async fn remount(&self) -> AnyResult<()> {
         self.holders.write().await.clear();
-        self.run().await
+        self.run(Default::default()).await
```

Comment:
This can lead to a problems. Why do you pass a default instance?

Reply:
It's actually do nothing, but yeah it's not clear. Now I use another solution

Reply:
Ok, approach with `Hooks` trait is much better